fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/MetricLoader.java

```diff
@@ -14,7 +14,10 @@
      * Load metrics and populate the metric dictionary.
      *
      * @param metricDictionary  The dictionary that will be loaded with metrics
+     *
+     * @deprecated in favor of loadMetricDictionary(MetricDictionary, DimensionDictionary)
```

Comment:
My thoughts on this are that we've come to the realization that some metric definitions require dimensions. The best way (in my opinion) to make this dependency clear is to pass the "up-stream" components into the downstream ones so that it's obvious that the dimensions need to be set up before the metrics. Deprecating (and eventually removing) the unclear method call takes us that direction.